🎨 Limit UI iteration (styles & functionality)

fairlighteth opened the pull request:
# Summary

- Adds column 'Market price'
- Adds column 'Executes at'
- Adds column 'Filled'
- Small order warning banner + column indicator (+ tooltip)
- Tooltip hover on 'Executes at' price for open orders, indicating how close to executing the order is.
- Dot indicator for 'how close to executing' the order is.
- Explainer banner for Dot indicator functionality + column header tooltip
- Misc. style improvements on the order table layout.
- Adding back a close-able info banner below the limit order widget container.

<img width="1740" alt="Screenshot 2023-03-07 at 18 07 53" src="https://user-images.githubusercontent.com/31534717/223510715-9b812576-97e9-4aca-bacf-358dd23d3b99.png">

elena-zh commented:
Hey @fairlighteth , just to clarify: should I focus here on styles only or test functionality as well?

elena-zh reviewed (changes requested):
Anyways, just focusing on UI issues: 
1. I don't see 'Filled' column on the mock up (will be replaced with Execution time?)? As far as I can understand, amount it the column are not correct and require rework. Am I right? 
![image](https://user-images.githubusercontent.com/70885163/223701232-6507dc9e-57e5-48d0-8236-3f068239202c.png)

2. Also, small amount banner requires logic  implementation as not it is permanently displayed
3. Where Learn more links should navigate? 
![image](https://user-images.githubusercontent.com/70885163/223701677-a7c2d748-65ea-4471-97ea-6074c01e464c.png)
![image](https://user-images.githubusercontent.com/70885163/223701702-13b8c2b8-aba3-4df8-a98b-f503c0060e3c.png)
4. And it would be great to add arrows for the Learn more links (to correspond to other links on the UI)
![arrow](https://user-images.githubusercontent.com/70885163/223701840-1f2dbd47-6977-47ab-b526-821cc871ecab.png)
5. Warning icons are not in 1 line 
![not in line](https://user-images.githubusercontent.com/70885163/223701931-b9040573-074d-450b-a05b-7a9b022f33aa.png)
6. When hover a mouse on an amount in the dark mode, a newly implemented tooltip is almost invisible. Focus goes to the full amount on hover. Van we highlight the new tooltip somehow? 
![diff to see](https://user-images.githubusercontent.com/70885163/223702210-95603096-f3f0-4cf0-8004-c00f6f005e3c.png)
7. Scroll on the page appears when Open orders tab is opened, screen width is less than 1600 px (it is more than 14" screen). Can we adjust the table somehow to avoid this scrollbar displaying in a desktop view?
![scroll](https://user-images.githubusercontent.com/70885163/223702537-7f31c8eb-e0e4-431d-b600-e97786d8f651.png)
8. Banner messages are shifted on a reduced screen
![reduced screen](https://user-images.githubusercontent.com/70885163/223702658-46c7103b-b0d2-47a3-8275-021d677ef06a.png)
9. In order to read the banner's message in a mobile view a view needs to scroll to the right. Can we show the ranges below the banner name in a mobile view? 
![image](https://user-images.githubusercontent.com/70885163/223702942-958d4688-0b57-4735-8215-2702d5fa85a6.png)
10. It is not related to the current PR, but I noticed that 10th page is displayed out of the screen area in a mobile view
![10 is out](https://user-images.githubusercontent.com/70885163/223703096-240eae94-b3b4-4d6a-8df9-f64ce3048cc9.png)

Thanks!

fairlighteth commented:
> Hey @fairlighteth , just to clarify: should I focus here on styles only or test functionality as well?

Both please.

elena-zh commented:
> > Hey @fairlighteth , just to clarify: should I focus here on styles only or test functionality as well?
> 
> Both please.

As for functionality: 
1. 'Small orders' banner is displayed all the time (when connected/disconnected/no token pair is selected/amount is not selected/price is not selected/amount is NOT small).  I think, we should show the banner when an amount is really small
2. Filled column on the Orders history tab shows not valid amounts
![image](https://user-images.githubusercontent.com/70885163/223764910-3045d886-5236-4090-a0e5-66f765139710.png)
3. Tooltips are hardcoded 
![image](https://user-images.githubusercontent.com/70885163/223765081-74bf704a-798d-4097-beb0-020357a14513.png)
![image](https://user-images.githubusercontent.com/70885163/223765177-08acb1ec-8048-4d74-9148-249a398ea1b7.png)
4. I can't see blue/green indicators (I hope they will be fixed in another PR).

shoom3301 commented on src/cow-react/modules/limitOrders/containers/LimitOrdersWidget/index.tsx:
```diff
@@ -292,6 +295,14 @@ const LimitOrders = React.memo((props: LimitOrdersProps) => {
                 </styledEl.FooterBox>
               )}
 
+              {/* TODO: Move this component inside LimitOrdersWarnings */}
```
Just remainder about TODO, fix it when you wish

shoom3301 commented on src/cow-react/modules/limitOrders/pure/InfoBanner/index.tsx:
```diff
@@ -0,0 +1,40 @@
+
+  const closeBanner = (): void => {
+    setShowBanner(false)
+    localStorage.setItem(LOCAL_STORAGE_KEY, 'false')
```
Do we need this? Bellow we have useEffect that does the same

fairlighteth replied:
Feel free to re-factor as I'm unsure what is needed or not.

shoom3301 commented on src/cow-react/modules/limitOrders/pure/Orders/OrderRow/index.tsx:
```diff
@@ -9,17 +9,24 @@ import { getSellAmountWithFee } from '@cow/modules/limitOrders/utils/getSellAmou
 import { getEtherscanLink } from 'utils'
 import { PendingOrderPrices } from '@cow/modules/orders/state/pendingOrdersPricesAtom'
-import Loader from 'components/Loader'
+import Loader from '@src/components/Loader'
```
```suggestion
import Loader from 'components/Loader'
```

shoom3301 reviewed (approved):
Soft approve

elena-zh commented:
Hey @fairlighteth , seems like the build is failed as I don't see changes in the PR

elena-zh commented:
Hey @fairlighteth , great job!
Some nitpicks from my side: 
1. Seems like dots have different size on the screen 1536x864
![diff size](https://user-images.githubusercontent.com/70885163/224132074-cc1136c4-ce20-4463-bf3c-151fff2b385e.png)
2. I still see warning icons shifted from the value on reduced screens
![reduced screen](https://user-images.githubusercontent.com/70885163/224132177-20893ec5-fb61-4cdf-ba7b-741ef411bf8d.png)
3. Scroll under the pagination in Win 11 does not look good + pagination is left-aligned now
![scrollbar](https://user-images.githubusercontent.com/70885163/224132352-07054a1c-970d-4594-b341-44a800f9e1d6.png)
4. Sometimes it is a bit difficult to tap on the warning icon in a mobile view (need to do it twice). But it is a nitpick. 


Thanks!

alfetopito commented:
@fairlighteth let's use the base branch `feature/limit-ui-update` for now on https://github.com/cowprotocol/cowswap/pull/2139

Contains everything in this PR plus latest develop.
Already addressed all the merge conflicts (there weren't many)

fairlighteth commented:
@elena-zh Thank you again for your review:

1. Fixed.
2. Fixed: I now just right align them by default.
3. This is intentional and by design. On actual mobile devices you will not see the scrollbar (you are testing a mobile viewport on a non-mobile environment)
4. Made it bigger now, which should make it easier to tap on.
5. Fixed.

alfetopito commented:
As agreed, merging to parent to consolidate.
Anything still pending will be addressed as follow up PR
